### PR TITLE
Add close method

### PR DESCRIPTION
**Pull request by lukeburns**
Depends on https://github.com/mafintosh/signalhub/pull/22 and https://github.com/mafintosh/webrtc-swarm/pull/14

### CODE REVIEW

**Comment by lukeburns** [comment]
https://github.com/karissa/hyperdrive-archive-swarm/pull/4 has conflicts

File: index.js [hunk 1]
```diff
@@ -29,18 +29,22 @@ function HyperdriveSwarm (archive, opts) {
...
+}
+
 HyperdriveSwarm.prototype._browser = function () {
   var self = this
   var swarmKey = (self.opts.signalhubPrefix || 'dat-') + self.archive.discoveryKey.toString('hex')
   self.browser = webRTCSwarm(signalhub(swarmKey, self.signalhub), {wrtc: self.opts.wrtc})
   self.browser.on('peer', function (conn) {
     var peer = self.archive.replicate()
     self.connections++
-    console.log(self.connections)
```
Comment (okdistribute) [inline index.js:39]: nice catch ;)

**Comment by okdistribute** [comment]
A+ as soon as the other PRs get merged and published

File: index.js (hunk 1 above)
Comment (lukeburns) [inline index.js:39]: :)

**Comment by lukeburns** [comment]
just waiting on https://github.com/mafintosh/webrtc-swarm/pull/14

**Comment by lukeburns** [comment]
Added standard to test - https://github.com/karissa/hyperdrive-archive-swarm/pull/12 has conflicts

**Comment by lukeburns** [comment]
https://github.com/mafintosh/webrtc-swarm/pull/14 was merged

File: example.js [hunk 2]
```diff
@@ -5,13 +5,13 @@ var swarm = require('.')
 var drive = hyperdrive(memdb())
 var archive = drive.createArchive()
 archive.finalize(function () {
+  console.log('public key', archive.key.toString('hex'))
```
Comment (okdistribute) [inline example.js]: lets remove these console.logs

File: index.js [hunk 3]
```diff
@@ -70,8 +87,6 @@ HyperdriveSwarm.prototype._node = function () {
   })
 
   swarm.listen(self.opts.port || 3282)
-  self.node = swarm
+  self.discoverySwarm = swarm
```
Comment (okdistribute) [inline index.js]: i dont think its necessary to change the API here. let's stay as backwards compatible as possible

**Comment by okdistribute** [comment]
great job

**Comment by lukeburns** [comment]
👍

**Comment by okdistribute** [comment]
published at 3.1.0

**Pull request merged**